nncf/torch/quantization/weights_compression.py

```diff
@@ -0,0 +1,90 @@
+from typing import Dict, Optional
+
+import torch
+from torch import nn
+
+from nncf.torch.layers import NNCF_WRAPPED_USER_MODULES_DICT
+from nncf.torch.layers import NNCFEmbedding
+from nncf.torch.layers import NNCFLinear
+from nncf.torch.quantization.quantize_functions import get_scale_zp_from_input_low_input_high
+
+
+class WeightsCompressor(nn.Module):
+    """Class provided fake quantize operation or dequantization of compressed weights in forward pass
```

Comment:
This class tries to handle 2 responsibilities.  
IMO, there will be more clear and robust code with 2 separate classes for a certain task. 
First - Dequantizer
Second - TorchFakeQuantizer